JetAnalyzers/bin/jet_response_fitter_x.cc

```diff
@@ -193,25 +210,25 @@ int main(int argc,char**argv)
         continue;
       }
 
-      //if(verbose>0) cout << "Attempting to fit " << histname << " ... " << endl;
-
-      //if (histname.find("RelRsp")>5&&histname.find("AbsRsp")!=0) {
-      //  hrsp->Write();
-      //  continue;
-      //}
-      
+      //
+      // start fit process
+      //
       if (hrsp->Integral()>0.0) {
         int fitstatus(0);
         if (0==fittype) fit_gaussian(hrsp,nsigma,jtptmin,niter,verbose);
-        else fitstatus = fit_dscb(hrsp,nsigma,jtptmin,niter,alg,verbose);
+        else if (1==fittype) fitstatus = fit_dscb(hrsp,nsigma,jtptmin,niter,alg,verbose);
+	else fit_modified_fnc_dscb(hrsp, alg, calibrated, histName);        
+
+        TF1* fitfnc{0};
+	if(fittype==2) fitfnc = (TF1*) hrsp->GetFunction("modified_fdscb");
+	else fitfnc = (TF1*) hrsp->GetListOfFunctions()->Last();
```

Comment:
I think the GetFunction part also falls under the category of figuring out why the GetListOfFunctions()->Last() doesn't work for you. Is it possible to look into this? Do you need my help to debug that.

I know you said you need it this way for now, but I'm afraid that if we merge it this way that it will never get fixed.